charts/consul/templates/gossip-encryption-autogeneration-role.yaml

```diff
@@ -0,0 +1,31 @@
+{{- if .Values.global.gossipEncryption.autoGenerate }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ template "consul.fullname" . }}-gossip-encryption-autogenerate
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "consul.name" . }}
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  annotations:
+    "helm.sh/hook": pre-install,pre-upgrade
+    "helm.sh/hook-delete-policy": before-hook-creation
```

Comment:
Same question as above about the `hook-succeeded` policy.

Reply:
Adding it would kill the role before the job starts. I understand where the call out comes from though.

Reply:
aha!! that makes sense! thanks for the clarification!!